qcodes/dataset/measurements.py

```diff
@@ -190,17 +194,23 @@ def add_result(self, *res_tuple: res_type) -> None:
                 if inserting_this_as_array:
                     raise ValueError("Trying to insert into an ArrayType with "
                                      "a scalar value")
-                if param_spec.type == 'text':
+                if param_spec.type != 'numeric':
                     raise ValueError(f"It is not possible to save a numeric "
                                      f"value for parameter {paramstr!r} "
                                      f"because its type class is "
-                                     f"'text', not 'numeric'.")
+                                     f"{param_spec.type!r}, not 'numeric'.")
             elif isinstance(value, str):
                 if param_spec.type != 'text':
                     raise ValueError(f"It is not possible to save a string "
```

Comment:
just wondering, do we have parametrization on existing param_types for tests for these exceptions? if so, are we sure that the tests are executed also for the new 'complex' paramtype ?

Reply:
I think we should completely rewrite this logic